android-activity/src/native_activity/glue.rs

```diff
@@ -75,18 +75,20 @@ pub enum State {
 
 #[derive(Debug)]
 pub struct WaitableNativeActivityState {
+    // TODO: Unnecessary NIHism: why not use ndk::NativeActivity?
```

Comment:
```suggestion
    // TODO: use ndk::NativeActivity?
```

I half guess there might have been a reason at the time, but don't really recall - it sounds like it would be fine to change.

Reply:
I half-guessed what was going on in the mean-time: the NDK crate that I inherited still has rough/incorrect edges in places I haven't been looking: https://github.com/rust-mobile/ndk/issues/478

And since afaik you ported this glue code straight from C, it was easier to append `unsafe { ffi::<original code> }` than rewriting it with `ndk` primitives.